lab_1/main.py

```diff
@@ -113,7 +201,37 @@ def detect_language_advanced(unknown_profile: dict, profiles: list, languages: l
     :param top_n: a number of the most common words
     :return: a language
     """
-    pass
+    if (not isinstance(unknown_profile, dict)
+            or not isinstance(profiles, list)
+            or not isinstance(languages, list)
+            or not isinstance(top_n, int)):
+        return None
+    proportions = []
+    languages_with_same_proportions = []
+    if len(languages) == 0:
+        for language_profile in profiles:
+            languages.append(language_profile.get('name'))
+    check = False
+    for language_name in languages:
+        for language_profile in profiles:
+            if language_name in language_profile.get('name'):
+                check = True
+    if not check:
+        return None
+    for language_profile in profiles:
+        if language_profile.get('name') in languages:
+            proportion = compare_profiles_advanced(unknown_profile, language_profile, top_n)
+            proportions.append([language_profile.get('name'), proportion.get('score')])
+    for res in proportions[:-1]:
+        if proportions[0][1] == proportions[proportions.index(res) + 1][1]:
+            languages_with_same_proportions.append(proportions[proportions.index(res)][0])
```

Comment:
hard to read, explain these lines, please

Reply:
In _proportions_ list we have lists with languages' _names_ and the _score_ of shared proportions. Then we compare _scores_ with the first one and if some languages have same _scores_, we write them in a related list to sort by _names_. But now I see that it actually makes no sense, so I'll add sorting by _scores_, thus we can compare the most probable score with the next ones.